text/0028-bill-of-materials.md

```diff
@@ -0,0 +1,49 @@
+# Implement a Bill of Materials Across Paketo
+
+## Summary
+
+A bill of materials (BOM) is a set of metadata that provides an account of the
+dependencies and tools that are both present in an application image, as well
+those used in the construction of that image. The availability of this data is
+a key value proposition of Cloud Native Buildpacks.
+
+As an implementation of Cloud Native Buildpacks, the Paketo project should
+support adding bill of materials metadata across the different components that
+go into an app image.
+
+## Motivation
+
+A BOM can include an array of different metadata about application dependencies
+and buildpack dependencies such as SHAs, URIs, versions, license information,
```

Comment:
I think we should try to standardize on the information we would ideally like to include in every entry (if possible) and the key names we will use. For example, It would be annoying for users if we used `url` in one buildpack and `uri` in another. I don't think this is likely to happen in practice, but using this RFC as a source of truth for the intended data format would make it more useful.

The list will not be comprehensive as there may be additional fields that are meaningful only in some ecosystems.

Reply:
+1 to what @ekcasey said

>The list will not be comprehensive as there may be additional fields that are meaningful only in some ecosystems.

I think the wording in the RFC should be such that it defines a *minimal* set of key/values that need to be included. Thus enabling individual buildpacks to include additional metadata that might be relevant to only its usecase.

Reply:
Address this in https://github.com/paketo-buildpacks/rfcs/pull/57/commits/59e94453a4f0878a26b708553b532e3713324902. Let me know if this wording is strong enough